subt_ign/src/path_tracer.cc

```diff
@@ -17,9 +17,166 @@
 #include "path_tracer.hh"
 
 //////////////////////////////////////////////////
-Processor::Processor(const std::string &_path, double _rtf)
-  : rtf(_rtf)
+// Load a color from YAML helper function.
+// \param[in] _node YAML node that contains color information
+// \param[in] _def Default color value that is used if a color property is
+// missing.
+ignition::math::Color loadColor(const YAML::Node &_node, double _def = 1.0)
 {
+  ignition::math::Color clr(_def, _def, _def, _def);
+
+  if (_node["r"])
+    clr.R(_node["r"].as<double>());
+
+  if (_node["g"])
+    clr.G(_node["g"].as<double>());
+
+  if (_node["b"])
+    clr.B(_node["b"].as<double>());
+
+  if (_node["a"])
+    clr.A(_node["a"].as<double>());
+
+  return clr;
+}
+
+//////////////////////////////////////////////////
+MarkerColor::MarkerColor(const YAML::Node &_node)
+{
+  if (_node["ambient"])
+    this->ambient = loadColor(_node["ambient"]);
+
+  if (_node["diffuse"])
+    this->diffuse = loadColor(_node["diffuse"]);
+
+  if (_node["emissive"])
+    this->emissive = loadColor(_node["emissive"]);
+}
+
+//////////////////////////////////////////////////
+MarkerColor::MarkerColor(const MarkerColor &_clr)
+  :ambient(_clr.ambient), diffuse(_clr.diffuse), emissive(_clr.emissive)
+{
+}
+
+//////////////////////////////////////////////////
+MarkerColor::MarkerColor(const ignition::math::Color &_ambient,
+    const ignition::math::Color &_diffuse,
+    const ignition::math::Color &_emissive)
+  : ambient(_ambient), diffuse(_diffuse), emissive(_emissive)
+{
+}
+
+//////////////////////////////////////////////////
+Processor::Processor(const std::string &_path, const std::string &_configPath)
+{
+  YAML::Node cfg;
+  if (!_configPath.empty())
+  {
+    if (ignition::common::exists(_configPath))
+    {
+      try
+      {
+        cfg = YAML::LoadFile(_configPath);
+      }
+      catch (...)
+      {
+        std::cerr << "Unable to load configuration file["
+          << _configPath << "]\n";
+      }
```

Comment:
```suggestion
      catch (YAML::Exception &ex)
      {
        std::cerr << "Unable to load configuration file["
          << _configPath << "]: "
          << "error at line " << ex.mark.line + 1 << ", column "
          << ex.mark.column + 1 << ": " << ex.msg << "\n";
      }
```

Forwards YAML parsing errors, for example:

```
Unable to load configuration file[/home/michael/workspaces/subt_ws/config.yaml]: error at line 8, column 3: end of map not found
```